library/public/types.h

```diff
@@ -58,6 +71,54 @@ class F3D_EXPORT ratio_t
   double Value;
 };
 
+/**
+ * Describe a RGB color.
+ */
+class F3D_EXPORT color_t
+{
+public:
+  color_t() = default;
+  color_t(const std::vector<double>& vec)
```

Comment:
```suggestion
  explicit color_t(const std::vector<double>& vec)
```

Reply:
actually I dont think it should be explicit.

Reply:
any reason?

Reply:
It would require to make some change in the cmake generation code logic and to rework the variant api in order to call the right constructor. I designed the option variant API with the idea that option type will be convertable seamlessly to one of 5 fundamental types, in that case, std::vector<double>.

Reply:
ok